From 7d02563a5b6f2770efb431c2efb27a8aba1001f3 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 10 Jun 2022 10:52:45 +0300
Subject: [PATCH] [SW-89971] rom: remove elbi axprot setting

PCIE ELBI interface is 'privileged' by default. Hence it was set to
'secured' in ROM. But for gaudi3, this fix is moved to RTL, so
the same can be removed from ROM.

Change-Id: Ie390852f19ee8e369385be966e3c0bb62204675c
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 729beb2..42e61f0 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -847,14 +847,6 @@
 	}
 }
 
-static void security_pcie_elbi_axprot_config(void)
-{
-	regmap_writel(AXPROT_UNPRIV_VAL, mmD0_PCIE_ELBI_RR_MSTR_IF_AXPROT_LBW_AWPROT_0_OVRD);
-	regmap_writel(AXPROT_UNPRIV_VAL, mmD0_PCIE_ELBI_RR_MSTR_IF_AXPROT_LBW_ARPROT_0_OVRD);
-	regmap_writel(AXPROT_SEC_VAL, mmD0_PCIE_ELBI_RR_MSTR_IF_AXPROT_LBW_AWPROT_1_OVRD);
-	regmap_writel(AXPROT_SEC_VAL, mmD0_PCIE_ELBI_RR_MSTR_IF_AXPROT_LBW_ARPROT_1_OVRD);
-}
-
 static void security_rr_recovery_unsecured_config(void)
 {
 	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
@@ -873,7 +865,6 @@
 void security_rr_recovery_config(void)
 {
 	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK) {
-		security_pcie_elbi_axprot_config();
 		security_prot_bits_config();
 		security_rr_recovery_lbw_config();
 		security_rr_recovery_hbw_config();
